### DISCUSSION TITLE
custom validation with custom mask

### DISCUSSION THREAD

**Discussion by Amin-Partovi** [react-hook-form/react-hook-form#6017] [Q&A]
I have issue with validation when I use my custom mask, I appreciate any help

here is my mask

`const mobileMask = value =>
  value
    .replace(/[۰-۹]/g, d => "۰۱۲۳۴۵۶۷۸۹".indexOf(d).toString())
    .replace(/[^\d]/g, "")
    .slice(0, 11);`

and here is my customValidation

`function MobileValidation(mobile) {
  if (mobile && !/^09[0-9]{9}$/.test(mobile)) {
    var error = (
      <span className={styles.error}>{translate.validation.wrongNumber}</span>
    );
  }
  return error;
}`

I set my validation and mask as following:
`<CustomInput
                name="mobileNumber"
                mandatory={true}
                error={errors.mobileNumber?.message}
                type="text"
                label={translate.addUser.mobile}
                touched={formState.touched[`mobileNumber`] ? true : false}
                register={register({
                  required: {
                    value: true,
                    message: FieldIsRequired(watch(`mobileNumber`)),
                  },
                  validate: value => MobileValidation(value),
                })}
                onChange={e =>
                  setValue(
                    `mobileNumber`,
                    mobileMask (e.target.value)
                  )
                }
              />`

if I console log the value parameter in validate I saw that the value did not follow the mask, however the value rendered in input followed the mask. these issue lead to error displayed whereas the value in input is correct

![123](https://user-images.githubusercontent.com/73663791/126270652-b4c65156-625b-4840-9639-2fa5427a8d20.png)

**Comment by bluebill1049** [comment]
I would recommend using `Controller` to wrap with your masked input and let the controller only do the validation.

https://react-hook-form.com/api/usecontroller/controller
